web/src/lib/nostr/utils.js

```diff
@@ -59,3 +57,39 @@ export function getChannelIdFromChannelName(channelName) {
 
     return getEventHash(event);
 }
+
+function createNostrKeys() {
+    // hex strings
+    let privateKey = generatePrivateKey()
+    let publicKey = getPublicKey(privateKey);
+
+    return {
+        'private': privateKey,
+        'public': publicKey,
+        'npub': nip19.npubEncode(publicKey)
+    };
+}
+
+export function getNostrKeysInitIfNecessary(user, token) {
+    if (user.nostr_private_key !== null) {
+        let publicKey = getPublicKey(user.nostr_private_key)
+        return {
+            'private': user.nostr_private_key,
+            'public': publicKey,
+            'npub': nip19.npubEncode(publicKey)
+        };
+    }
+
+    let nostrKeys = createNostrKeys();
+    let nostr_private_key = nostrKeys.private;
+
+    putProfile(token, {nostr_private_key},
```

Comment:
i don't like a few things here:

1) a function called "get..." calls the API with a PUT

and more generally...

2) i don't like the fact that the "nostr utils" package is accessing our API. they should be unrelated!

better to just return something to let the caller (svelte component) know that the profile should be updated.

Reply:
You're right. When I did this a month ago, I was calling this from several places, but anyways it's better in the component.